### ISSUE TITLE
Doc comments on ValueEnum case objects make scaladoc fail

### ISSUE THREAD

**Issue by hypatian** [lloydmeta/enumeratum#128]
If you create a ValueEnum class which has doc comments on the ValueEnumEntry case classes, it will build just fine but scaladoc will fail with, for example: "looks like not all of the members have a literal/constant 'value:int' declaration, namely: object Foo."

My best guess is that scalac and scaladoc use the same AST, but scalac discards doc comment nodes early and scaladoc does not. When ValueEnumMacros.findValuesForSubclassTrees starts parsing through the ValueEnum, the extra unexpected AST nodes block its attempt to figure out the values for the cases.

The following code will demonstrate the problem. If you remove the doc comment or move it from Foo to values, scaladoc will succeed. But with the doc comment on Foo, it will fail.

```scala
import enumeratum.values.{ShortEnum, ShortEnumEntry}
sealed abstract class Test(val value: Short) extends ShortEnumEntry
case object Test extends ShortEnum[Test] {
  /** Blah blah blah */
  case object Foo extends Test(0)
  val values = findValues
}
```

This was observed using Scala 2.11.8 and enumeratum 1.5.8.
(Also further checked against Scala 2.12.1, where the behavior seems the same.)

**Comment by lloydmeta** [comment]
Hi there,

Thanks for filing this issue :) The main cause looks to be an open issue with Scala filed as [SI-8904](https://issues.scala-lang.org/browse/SI-8904).

I've found a work around for it and will hopefully be able to push a fix out soon.

**Comment by lloydmeta** [comment]
Someone reported a similar issue in MacWire along with a [workaround](https://github.com/adamw/macwire/issues/21#issuecomment-58800161) (I've not tested it yet)

> You can ommit compiler bug with simply adding following setting to your build.sbt:
>
> `scalacOptions in (Compile,doc) ++= Seq("-Ymacro-expand:none")`
>
> Above line will disable macro expansion during compile:doc.

**Comment by lloydmeta** [comment]
Closed via #129
